### DISCUSSION TITLE
How to hide shadows

### DISCUSSION THREAD

**Discussion by bamiendev** [google/model-viewer#4639] [Q&A]
![z5070147661063_303c40d91b547ffa258c4489c6efe705](https://github.com/google/model-viewer/assets/116520283/91df4485-5d54-40c7-97d6-11b97cc5ae9e)

I have hidden the materials by:
setAlphaMode("MASK")
setAlphaCutoff(2)

How to hide shadows of all set materials:
setAlphaMode("MASK")
setAlphaCutoff(2)

**Comment by danilokrulj** [comment]
Don't hide the objects like that, find them in the hierarchy and set their visibility to false. Than their shadows should disappear.

**Reply by bamiendev** [reply]
How to make visibility it false, it's not in document

**Reply by bamiendev** [reply]
Aaa here is more api of three.js, it's too complicated for me. I need help, please help me

**Reply by danilokrulj** [reply]
Use something like:

    const mv = document.querySelector("model-viewer");
    const sceneSym = Object.getOwnPropertySymbols(mv).find(x => x.description === "scene");
    
    const model1 = modelViewer[sceneSym].children[0].getObjectByName('model1');
    const model2 = modelViewer[sceneSym].children[0].getObjectByName('model2');
    const model3 = modelViewer[sceneSym].children[0].getObjectByName('model3');
    const model4  = modelViewer[sceneSym].children[0].getObjectByName('model4 ');
    
    model1.visible = false;
    
and so on, you just need to find the path to the correct meshes... cant give you exact code because i dont know the structure of the model

**Reply by danilokrulj** [reply]
Btw this is code from some older project, not sure it works 100% but should give you an idea how to do the thing you want to do.

**Reply by papipsycho** [reply]
is the same type of code i use, is working, i mean it's hide the object, but is still display the shadow, if anyone have an idea ?

**Reply by danilokrulj** [reply]
Ahh I forgot to add this... you need to update the shadow after you hide/show objects. 

modelViewer[sceneSym].updateShadow();
modelViewer[sceneSym].queueRender();
